jenkinsbuilder/cccp_index_reader.py

```diff
@@ -183,52 +185,36 @@ def get_old_project_list():
                      projects_list)
         logger.error("I/O Error {0}:{1}".format(e.errno, e.strerror))
         return []
-    except:
+    except Exception as e:
         logger.error("Failed to read project names from file: %s",
                      projects_list)
         logger.error("Unexpected error: %s", sys.exc_info()[0])
         return []
 
 
-def find_stale_projects(old, new):
-    """
-    This function diffs the old project list with new project
-    list. And returns the list of project names to be deleted
-    """
-    return list(set(old) - set(new))
-
-
-def delete_stale_projects_on_jenkins(stale_projects):
+def get_new_project_list(indexdlocation):
     """
-    This function deletes the stale entries at jenkins that
-    are no longer valid. If there is any issue/exception, it
-    bypasses by printing the error.
+    This function returns list of projects in the container-index. It's called
+    get_new_project_list because it takes a fresh look at container-index every
+    time. It requires a path to the directory containing container-index and
+    returns a list
     """
-    for project in stale_projects:
-        myargs = ["jenkins-jobs", "delete", project]
-        # print either output or error
-        _, error = run_command(myargs)
-        if error:
-            logger.critical("Failed to delete project %s. Exiting..", project)
-            logger.critical(sys.exc_info()[0])
-            exit(1)
-            # if a job fails to be deleted from jenkins it will create issues
-            # cccp-index job at jenkins needs to fail and be notified
-    logger.info("Deleted stale projects successfully.")
+    new_projects_names = []
+    for project in get_projects_from_index(indexdlocation):
+        new_projects_names.append(
+            str(project[0]["project"]["appid"]) + "/" +
+            str(project[0]["project"]["jobid"]) + ":" +
+            str(project[0]["project"]["desired_tag"])
+        )
+    return new_projects_names
 
 
-def run_command(command):
+def create_or_update_project_on_jenkins(indexdlocation):
```

Comment:
just append the new functions to the script, if we insert new functions in between it's confusing and difficult to review.